pkg/setup/setup.go

```diff
@@ -167,10 +166,10 @@ func (s *setup) listenToShutdownSignal(ctx context.Context, cancelFunc context.C
 	cancelFunc()
 
 	// call shutdown hooks
-	gracefulShutdown(ctx, s, 30*time.Second)
+	gracefulShutdown(s, 30*time.Second)
 }
 
-func gracefulShutdown(ctx context.Context, s *setup, timeout time.Duration) {
+func gracefulShutdown(s *setup, timeout time.Duration) {
 	// Instantiate background context
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
Uh, that is a good catch. I would think that the ctx paramer actually makes sense and we should not use `context.Background()` but `ctx` as first pararmeter here to pass the logger and tracer from the context.

Reply:
What I understood from tracing the code is that we don't actually want to use ctx argument as it is probably already cancelled previously, but I might be wrong here.

Reply:
I haven't tested it, but I would assume that still use a context for logging, even when it's cancelled. In which case we should do what hannes suggested.